src/main/java/fruitshop/strategy/SupplyOperation.java

```diff
@@ -0,0 +1,18 @@
+package fruitshop.strategy;
+
+import fruitshop.db.Storage;
+import fruitshop.model.FruitTransaction;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void handler(FruitTransaction fruitTransaction) {
+        Storage storage = new Storage();
+        int newQuantity = fruitTransaction.getQuantity()
+                + storage.getQuantity(fruitTransaction.getFruit());
+        if (newQuantity >= 0) {
+            storage.set(fruitTransaction.getFruit(), newQuantity);
```

Comment:
The check for `newQuantity >= 0` is unnecessary here since the supply operation is expected to only add to the existing quantity and should not result in a negative number. You can remove this check.